src/common/direct-link-query.js

```diff
@@ -27,6 +38,11 @@ function directLinkQuery(url) {
     return { query };
   }
 
+  var groupMatch = url.match(/#annotations:group:(.*)$/);
```

Comment:
This should be matching url-safe-base64 identifiers rather than `.*`.

Reply:
I had another look at pubids in h. It turns out they are different to annotation IDs. They are currently a subset of alphanumeric characters designed to avoid certain (English-language) readability issues: https://github.com/hypothesis/h/blob/b4944f117232fd4eb497b72e1ddd4f5879e974fe/h/pubid.py#L6